public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/Advanced.test.tsx

```diff
@@ -0,0 +1,35 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import Advanced from './Advanced';
+
+describe('AzureMonitor ResourcePicker', () => {
+  it('should set a parameter as an object', async () => {
+    const onChange = jest.fn();
+    const { rerender } = render(<Advanced onChange={onChange} resource={{}} />);
+    const advancedSection = screen.getByText('Advanced');
+    advancedSection.click();
+
+    const subsInput = await screen.findByLabelText('Subscription');
+    await userEvent.type(subsInput, 'd');
+    expect(onChange).toHaveBeenCalledWith({ subscription: 'd' });
+
+    rerender(<Advanced onChange={onChange} resource={{ subscription: 'def-123' }} />);
+    expect(screen.getByDisplayValue('def-123')).toBeInTheDocument();
```

Comment:
Could we be more restrictive, using  get by label `Subscription` instead? to make sure we don't fill out the form incorrectly? (i.e. if we did not wire our code correctly, and somehow filled the resource group text field with the subscription value, this test will still pass)